bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java

```diff
@@ -631,6 +683,39 @@ public boolean requestUpdates(int requestCount, boolean refreshSettings) {
         return false;
     }
 
+    /**
+     * Map input states to channels
+     *
+     * @param groupName Channel Group (relay / relay1...)
+     *
+     * @param status Shelly device status
+     * @return true: one or more inputs were updated
+     */
+    @SuppressWarnings({ "null", "unused" })
```

Comment:
Why do you have to suppress unused? Can't you simply remove unused variables?
Also, please try to avoid null suppressions and rather refactor the code to be fully null-safe.
Ok, if you address this in a future PR, no blocker wrt merging now.

Reply:
because Eclipse sometimes just report BS

Reply:
example: For this statement Eclipse reports a redundant null check

`
Strong thingName = properties.get(PROPERTY_SERVICE_NAME) != null ? properties.get(PROPERTY_SERVICE_NAME).toLowerCase()
                : thingType;
`
how to resolved those warnings?

Reply:
Which is correct, because theoretically another thread could meanwhile remove the property before you call the get.
This should solve those issues:
```
String thingName = properties.get(PROPERTY_SERVICE_NAME);
thingName = thingName!=null ? thingName.toLowerCase() : thingType;
```

Reply:
nope, not it's "thingName!=null"

Reply:
Then probably this way:
```
```
String serviceName = properties.get(PROPERTY_SERVICE_NAME);
String thingName = serviceName!=null ? serviceName.toLowerCase() : thingType;
```
That's anyhow cleaner than my previous suggestion.